src/components/header/search.jsx

```diff
@@ -1,6 +1,8 @@
+import { lazy } from 'preact-iso';
 import style from './style.module.css';
 import config from '../../config.json';
-import { DocSearch } from '@docsearch/react';
+
+const DocSearch = lazy(() => import('@docsearch/react').then(m => m.DocSearch));
```

Comment:
Do we want a separate Suspenseful boundary for this so it doesn't bubble up to the route boundary?

Reply:
Could do, yeah.

I'm actually going to be away for the next few days (gotta get to bed now) so if you (or anyone else) wanted to add on top here certainly feel free. Else I'll hopefully get back to this in a few.